[CI] EsqlSecurityIT testLookupJoinIndexAllowed failing

**Build Scans:**
- [elasticsearch-periodic #5619 / release-tests](https://gradle-enterprise.elastic.co/s/owtpltdzupfjk)
- [elasticsearch-pull-request #48345 / release-tests](https://gradle-enterprise.elastic.co/s/rsbaj2wuypkj6)
- [elasticsearch-periodic #5614 / release-tests](https://gradle-enterprise.elastic.co/s/cnqvqf4ivc4o4)
- [elasticsearch-periodic #5609 / release-tests](https://gradle-enterprise.elastic.co/s/wtu2lyctthhe2)
- [elasticsearch-periodic #5604 / release-tests](https://gradle-enterprise.elastic.co/s/wbbx3zrehnjj2)
- [elasticsearch-periodic #5594 / release-tests](https://gradle-enterprise.elastic.co/s/u2mrnzqyujyta)
- [elasticsearch-periodic #5589 / release-tests](https://gradle-enterprise.elastic.co/s/6ayoyowb7hq7g)
- [elasticsearch-periodic #5584 / release-tests](https://gradle-enterprise.elastic.co/s/mvbg5jh45xziu)
- [elasticsearch-periodic #5579 / release-tests](https://gradle-enterprise.elastic.co/s/ieceyyr6fn2em)
- [elasticsearch-periodic #5564 / release-tests](https://gradle-enterprise.elastic.co/s/wuor4offgusuc)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:security:javaRestTest" --tests "org.elasticsearch.xpack.esql.EsqlSecurityIT.testLookupJoinIndexAllowed" -Dtests.seed=7BAFAE446F92AE00 -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key -Dtests.locale=en-AU -Dtests.timezone=PST -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.EsqlSecurityIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testLookupJoinIndexAllowed),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
org.elasticsearch.client.ResponseException: method [POST], host [http://[::1]:44601], URI [_query?error_trace=true], status line [HTTP/1.1 400 Bad Request]
{"error":{"root_cause":[{"type":"parsing_exception","reason":"line 1:33: mismatched input 'LOOKUP' expecting {'dissect', 'drop', 'enrich', 'eval', 'grok', 'keep', 'limit', 'mv_expand', 'rename', 'sort', 'stats', 'where'}","stack_trace":"org.elasticsearch.xpack.esql.parser.ParsingException: line 1:33: mismatched input 'LOOKUP' expecting {'dissect', 'drop', 'enrich', 'eval', 'grok', 'keep', 'limit', 'mv_expand', 'rename', 'sort', 'stats', 'where'}\n\tat org.elasticsearch.xpack.esql.parser.EsqlParser$1.syntaxError(EsqlParser.java:140)\n\tat org.antlr.v4.runtime.ProxyErrorListener.syntaxError(ProxyErrorListener.java:41)\n\tat org.antlr.v4.runtime.Parser.notifyErrorListeners(Parser.java:544)\n\tat org.antlr.v4.runtime.DefaultErrorStrategy.reportInputMismatch(DefaultErrorStrategy.java:327)\n\tat org.antlr.v4.runtime.DefaultErrorStrategy.reportError(DefaultErrorStrategy.java:139)\n\tat org.elasticsearch.xpack.esql.parser.EsqlBaseParser.processingCommand(EsqlBaseParser.java:651)\n\tat org.elasticsearch.xpack.esql.parser.EsqlBaseParser.query(EsqlBaseParser.java:347)\n\tat org.elasticsearch.xpack.esql.parser.EsqlBaseParser.singleStatement(EsqlBaseParser.java:230)\n\tat org.elasticsearch.xpack.esql.parser.EsqlParser.invokeParser(EsqlParser.java:96)\n\tat org.elasticsearch.xpack.esql.parser.EsqlParser.createStatement(EsqlParser.java:63)\n\tat org.elasticsearch.xpack.esql.session.EsqlSession.parse(EsqlSession.java:282)\n\tat org.elasticsearch.xpack.esql.session.EsqlSession.execute(EsqlSession.java:158)\n\tat org.elasticsearch.xpack.esql.execution.PlanExecutor.esql(PlanExecutor.java:83)\n\tat org.elasticsearch.xpack.esql.plugin.TransportEsqlQueryAction.innerExecute(TransportEsqlQueryAction.java:191)\n\tat org.elasticsearch.xpack.esql.plugin.TransportEsqlQueryAction.doExecuteForked(TransportEsqlQueryAction.java:153)\n\tat org.elasticsearch.xpack.esql.plugin.TransportEsqlQueryAction.lambda$doExecute$0(TransportEsqlQueryAction.java:137)\n\tat org.elasticsearch.server@9.0.0/org.elasticsearch.action.ActionRunnable$4.doRun(ActionRunnable.java:101)\n\tat org.elasticsearch.server@9.0.0/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)\n\tat org.elasticsearch.server@9.0.0/org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:34)\n\tat org.elasticsearch.server@9.0.0/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:1044)\n\tat org.elasticsearch.server@9.0.0/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\n\tat java.base/java.lang.Thread.run(Thread.java:1575)\nCaused by: org.antlr.v4.runtime.InputMismatchException\n\tat org.antlr.v4.runtime.DefaultErrorStrategy.sync(DefaultErrorStrategy.java:270)\n\tat org.elasticsearch.xpack.esql.parser.EsqlBaseParser.processingCommand(EsqlBaseParser.java:534)\n\t... 18 more\n"}],"type":"parsing_exception","reason":"line 1:33: mismatched input 'LOOKUP' expecting {'dissect', 'drop', 'enrich', 'eval', 'grok', 'keep', 'limit', 'mv_expand', 'rename', 'sort', 'stats', 'where'}","caused_by":{"type":"input_mismatch_exception","reason":null,"stack_trace":"org.antlr.v4.runtime.InputMismatchException\n\tat org.antlr.v4.runtime.DefaultErrorStrategy.sync(DefaultErrorStrategy.java:270)\n\tat org.elasticsearch.xpack.esql.parser.EsqlBaseParser.processingCommand(EsqlBaseParser.java:534)\n\tat org.elasticsearch.xpack.esql.parser.EsqlBaseParser.query(EsqlBaseParser.java:347)\n\tat org.elasticsearch.xpack.esql.parser.EsqlBaseParser.singleStatement(EsqlBaseParser.java:230)\n\tat org.elasticsearch.xpack.esql.parser.EsqlParser.invokeParser(EsqlParser.java:96)\n\tat org.elasticsearch.xpack.esql.parser.EsqlParser.createStatement(EsqlParser.java:63)\n\tat org.elasticsearch.xpack.esql.session.EsqlSession.parse(EsqlSession.java:282)\n\tat org.elasticsearch.xpack.esql.session.EsqlSession.execute(EsqlSession.java:158)\n\tat org.elasticsearch.xpack.esql.execution.PlanExecutor.esql(PlanExecutor.java:83)\n\tat org.elasticsearch.xpack.esql.plugin.TransportEsqlQueryAction.innerExecute(TransportEsqlQueryAction.java:191)\n\tat org.elasticsearch.xpack.esql.plugin.TransportEsqlQueryAction.doExecuteForked(TransportEsqlQueryAction.java:153)\n\tat org.elasticsearch.xpack.esql.plugin.TransportEsqlQueryAction.lambda$doExecute$0(TransportEsqlQueryAction.java:137)\n\tat org.elasticsearch.server@9.0.0/org.elasticsearch.action.ActionRunnable$4.doRun(ActionRunnable.java:101)\n\tat org.elasticsearch.server@9.0.0/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)\n\tat org.elasticsearch.server@9.0.0/org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:34)\n\tat org.elasticsearch.server@9.0.0/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:1044)\n\tat org.elasticsearch.server@9.0.0/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\n\tat java.base/java.lang.Thread.run(Thread.java:1575)\n"},"stack_trace":"org.elasticsearch.xpack.esql.parser.ParsingException: line 1:33: mismatched input 'LOOKUP' expecting {'dissect', 'drop', 'enrich', 'eval', 'grok', 'keep', 'limit', 'mv_expand', 'rename', 'sort', 'stats', 'where'}\n\tat org.elasticsearch.xpack.esql.parser.EsqlParser$1.syntaxError(EsqlParser.java:140)\n\tat org.antlr.v4.runtime.ProxyErrorListener.syntaxError(ProxyErrorListener.java:41)\n\tat org.antlr.v4.runtime.Parser.notifyErrorListeners(Parser.java:544)\n\tat org.antlr.v4.runtime.DefaultErrorStrategy.reportInputMismatch(DefaultErrorStrategy.java:327)\n\tat org.antlr.v4.runtime.DefaultErrorStrategy.reportError(DefaultErrorStrategy.java:139)\n\tat org.elasticsearch.xpack.esql.parser.EsqlBaseParser.processingCommand(EsqlBaseParser.java:651)\n\tat org.elasticsearch.xpack.esql.parser.EsqlBaseParser.query(EsqlBaseParser.java:347)\n\tat org.elasticsearch.xpack.esql.parser.EsqlBaseParser.singleStatement(EsqlBaseParser.java:230)\n\tat org.elasticsearch.xpack.esql.parser.EsqlParser.invokeParser(EsqlParser.java:96)\n\tat org.elasticsearch.xpack.esql.parser.EsqlParser.createStatement(EsqlParser.java:63)\n\tat org.elasticsearch.xpack.esql.session.EsqlSession.parse(EsqlSession.java:282)\n\tat org.elasticsearch.xpack.esql.session.EsqlSession.execute(EsqlSession.java:158)\n\tat org.elasticsearch.xpack.esql.execution.PlanExecutor.esql(PlanExecutor.java:83)\n\tat org.elasticsearch.xpack.esql.plugin.TransportEsqlQueryAction.innerExecute(TransportEsqlQueryAction.java:191)\n\tat org.elasticsearch.xpack.esql.plugin.TransportEsqlQueryAction.doExecuteForked(TransportEsqlQueryAction.java:153)\n\tat org.elasticsearch.xpack.esql.plugin.TransportEsqlQueryAction.lambda$doExecute$0(TransportEsqlQueryAction.java:137)\n\tat org.elasticsearch.server@9.0.0/org.elasticsearch.action.ActionRunnable$4.doRun(ActionRunnable.java:101)\n\tat org.elasticsearch.server@9.0.0/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)\n\tat org.elasticsearch.server@9.0.0/org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:34)\n\tat org.elasticsearch.server@9.0.0/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:1044)\n\tat org.elasticsearch.server@9.0.0/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\n\tat java.base/java.lang.Thread.run(Thread.java:1575)\nCaused by: org.antlr.v4.runtime.InputMismatchException\n\tat org.antlr.v4.runtime.DefaultErrorStrategy.sync(DefaultErrorStrategy.java:270)\n\tat org.elasticsearch.xpack.esql.parser.EsqlBaseParser.processingCommand(EsqlBaseParser.java:534)\n\t... 18 more\n"},"status":400}
```

**Issue Reasons:**
- [main] 10 consecutive failures in step release-tests
- [main] 10 failures in test testLookupJoinIndexAllowed (3.3% fail rate in 302 executions)
- [main] 10 failures in step release-tests (100.0% fail rate in 10 executions)
- [main] 9 failures in pipeline elasticsearch-periodic (90.0% fail rate in 10 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -251,18 +251,6 @@ tests:
   issue: https://github.com/elastic/elasticsearch/issues/119179
 - class: org.elasticsearch.smoketest.SmokeTestMultiNodeClientYamlTestSuiteIT
   issue: https://github.com/elastic/elasticsearch/issues/119191
-- class: org.elasticsearch.xpack.esql.EsqlSecurityIT
-  method: testLookupJoinIndexAllowed
-  issue: https://github.com/elastic/elasticsearch/issues/119268
-- class: org.elasticsearch.xpack.esql.EsqlSecurityIT
-  method: testLookupJoinIndexForbidden
-  issue: https://github.com/elastic/elasticsearch/issues/119269
-- class: org.elasticsearch.xpack.esql.EsqlAsyncSecurityIT
-  method: testLookupJoinIndexForbidden
-  issue: https://github.com/elastic/elasticsearch/issues/119270
-- class: org.elasticsearch.xpack.esql.EsqlAsyncSecurityIT
-  method: testLookupJoinIndexAllowed
-  issue: https://github.com/elastic/elasticsearch/issues/119271
 - class: org.elasticsearch.xpack.logsdb.qa.LogsDbVersusLogsDbReindexedIntoStandardModeChallengeRestIT
   method: testEsqlTermsAggregationByMethod
   issue: https://github.com/elastic/elasticsearch/issues/119355

